include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -0,0 +1,900 @@
+			readyForNorm.barrier.srcAccessMask = asset::EAF_SHADER_WRITE_BIT;
+			readyForNorm.barrier.dstAccessMask = static_cast<asset::E_ACCESS_FLAGS>(asset::EAF_SHADER_READ_BIT);
+			readyForNorm.oldLayout = asset::EIL_GENERAL;
+			readyForNorm.newLayout = asset::EIL_SHADER_READ_ONLY_OPTIMAL;
+			readyForNorm.srcQueueFamilyIndex = ~0u;
+			readyForNorm.dstQueueFamilyIndex = ~0u;
+			readyForNorm.image = normalizationInImage;
+			readyForNorm.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+			readyForNorm.subresourceRange.levelCount = 1u;
+			readyForNorm.subresourceRange.layerCount = normalizationInImage->getCreationParameters().arrayLayers;
+			cmdbuf->pipelineBarrier(asset::EPSF_COMPUTE_SHADER_BIT, asset::EPSF_COMPUTE_SHADER_BIT, asset::EDF_NONE, 0u, nullptr, 1u, &alphaTestBarrier, 1u, &readyForNorm);
+
+			cmdbuf->bindDescriptorSets(asset::EPBP_COMPUTE, normalizationPipeline->getLayout(), 0u, 1u, &normalizationDS);
+			cmdbuf->bindComputePipeline(normalizationPipeline);
+			dispatchHelper(cmdbuf, normalizationPipeline->getLayout(), pushConstants, dispatchInfo);
+		}
+	}
+
+	//! WARNING: This function blocks and stalls the GPU!
+	template <typename KernelX, typename KernelY, typename KernelZ, typename... Args>
+	inline void blit(video::IGPUQueue* computeQueue, Args&&... args)
+	{
+		auto cmdpool = m_device->createCommandPool(computeQueue->getFamilyIndex(), video::IGPUCommandPool::ECF_NONE);
+		core::smart_refctd_ptr<video::IGPUCommandBuffer> cmdbuf;
+		m_device->createCommandBuffers(cmdpool.get(), video::IGPUCommandBuffer::EL_PRIMARY, 1u, &cmdbuf);
+
+		auto fence = m_device->createFence(video::IGPUFence::ECF_UNSIGNALED);
+
+		cmdbuf->begin(video::IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+		blit(cmdbuf.get(), std::forward<Args>(args)...);
+		cmdbuf->end();
+
+		video::IGPUQueue::SSubmitInfo submitInfo = {};
+		submitInfo.commandBufferCount = 1u;
+		submitInfo.commandBuffers = &cmdbuf.get();
+		computeQueue->submit(1u, &submitInfo, fence.get());
+
+		m_device->blockForFences(1u, &fence.get());
+	}
+
+	//! Returns the original format if supports STORAGE_IMAGE otherwise returns a format in its compat class which supports STORAGE_IMAGE.
+	inline asset::E_FORMAT getOutImageViewFormat(const asset::E_FORMAT format)
+	{
+		const auto& formatUsages = m_device->getPhysicalDevice()->getImageFormatUsagesOptimal(format);
+
+		if (formatUsages.storageImage && format != asset::EF_B10G11R11_UFLOAT_PACK32)
+		// if (formatUsages.storageImage) // RMSE: 0.227089
+		{
+			return format;
+		}
+		else
+		{
+			const asset::E_FORMAT compatFormat = getCompatClassFormat(format);
+
+			const auto& compatClassFormatUsages = m_device->getPhysicalDevice()->getImageFormatUsagesOptimal(compatFormat);
+			if (!compatClassFormatUsages.storageImage)
+				return asset::EF_UNKNOWN;
+			else
+				return compatFormat;
+		}
+	}
+
+	static inline asset::E_FORMAT getIntermediateFormat(const asset::E_FORMAT format)
```

Comment:
more descriptive name would be good like `getCoverageAdjustmentIntermediateFormat`

Reply:
Resolved.